google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/ProjectSelector.java

```diff
@@ -546,4 +596,36 @@ public void setModelNeedsRefresh(boolean modelNeedsRefresh) {
       this.modelNeedsRefresh = modelNeedsRefresh;
     }
   }
+
+  /**
+   * Interface that must be implemented in order to be informed of
+   * {@link ProjectSelectionChangedEvent} events.
+   */
+  public interface ProjectSelectionListener {
```

Comment:
That's a good idea, though I'm not really sure what is worth pulling up into the parent class.  `CustomizableComboBox` already has a way of listening on selection changes to tree nodes, using swing `TreeSelectionListeners`.

The main value of the ProjectSelectionListener interface is that it provides events that are data-aware - meaning it's provides the `Project` object and the user's `Credential`directly to the listeners, which it is able to provide since it's aware of the tree structure of the project selector's data model. 

I don't think this would make sense to move to a higher level of abstraction.